queue/driver_mongo.go

```diff
@@ -393,10 +396,21 @@ func (d *mongoDriver) reportingIndexes() []mongo.IndexModel {
 	return indexes
 }
 
-func (d *mongoDriver) Close() {
-	if d.canceler != nil {
-		d.canceler()
+// Close closes the resources for the driver. If it initialized a new client, it
+// is disconnected.
+func (d *mongoDriver) Close(ctx context.Context) error {
+	d.mu.Lock()
+	defer d.mu.Unlock()
```

Comment:
Should this check if Close was already called?

Reply:
I added a `closed` check. It has the same issue as the close no-op above, but I don't think it's worth the effort to make `Close` more robust.